### PR TITLE
[Communication] - Use Azure.Core.AccessToken instead of custom type

### PR DESCRIPTION
**Pull request by RezaJooyandeh**
ApiView: https://apiview.dev/Assemblies/Review/290c367d508340a386fc4a42d7474f36

We were using `CommunicationUserToken` that used to contain both an access token and a user id. But now our issue token route only returns the token so we can switch to `Azure.Core.AccessToken`.

Now that the type `CommunicationUserToken` is not exposed, I also removed the autorest extension that was renaming it from `CommunicationIdentityAccessToken`

I also removed some of the files that should have been removed in #18066

### CODE REVIEW

File: sdk/communication/Azure.Communication.Identity/CHANGELOG.md [hunk 1]
```diff
@@ -6,9 +6,12 @@
 - Added CommunicationIdentityClient (originally was part of the Azure.Communication.Administration package).
 - Added support to create CommunicationIdentityClient with TokenCredential.
 - Added support to create CommunicationIdentityClient with AzureKeyCredential.
+- Added ability to create a user and issue token for it at the same time.
 
 ### Breaking
 - CommunicationTokenScope.Pstn is removed.
+- Revoking tokens now revoke all the currently issued tokens instead of revoking tokens issued prior to a given time.
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.Identity/CHANGELOG.md:13]: Suggest changing this to CommunicationIdentityClient.RevokeTokens to match line 14.

File: sdk/communication/Azure.Communication.Identity/src/autorest.md [hunk 2]
```diff
@@ -7,6 +7,6 @@ Run `dotnet msbuild /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    -  https://raw.githubusercontent.com/Azure/azure-rest-api-specs/3e973fa7cd38a218772b1f5960d60b21da365239/specification/communication/data-plane/Microsoft.CommunicationServicesIdentity/stable/2021-03-07/CommunicationIdentity.json
+    -  https://raw.githubusercontent.com/Azure/azure-rest-api-specs/5b19c6e69cd2bb9dbe4e5c1237b2c5a175d90ca5/specification/communication/data-plane/Microsoft.CommunicationServicesIdentity/stable/2021-03-07/CommunicationIdentity.json
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Identity/src/autorest.md:10]: Which commit is the one we should use officially?

JS is using yet another commit it seems.

File: sdk/communication/Azure.Communication.Identity/README.md [hunk 3]
```diff
@@ -66,7 +66,7 @@ Console.WriteLine($"User id: {user.Id}");
 ### Issuing or Refreshing a token for an existing identity
```
Comment (minnieliu) [inline sdk/communication/Azure.Communication.Identity/README.md:66]: ...for an existing **user**

File: sdk/communication/Azure.Communication.Identity/src/autorest.md (hunk 2 above)
Comment (RezaJooyandeh) [reply sdk/communication/Azure.Communication.Identity/src/autorest.md:10]: `5b19c6e69cd2bb9dbe4e5c1237b2c5a175d90ca5` seems to be the latest commit that got merged 2 days ago

**Review by minnieliu — APPROVED** [review]

**Pull request merged**